RadixWallet/Clients/ProfileStore/ProfileStore.swift

```diff
@@ -185,100 +78,142 @@ extension ProfileStore {
+		@Dependency(\.device) var device
+		if !profile.hasMainnetAccounts() {
+			let errorMsg = "Incorrect implementation should have accounts on mainnet after finishing onboarding."
+			loggerGlobal.error(.init(stringLiteral: errorMsg))
+			assertionFailure(errorMsg)
+		}
+		let model = await device.model
+		let name = await device.name
+		let deviceDescription = DeviceInfo.deviceDescription(
+			name: name,
+			model: model
+		)
+		deviceInfo.description = deviceDescription
+		let lastUsedOnDevice = deviceInfo
+		try? secureStorageClient.saveDeviceInfo(lastUsedOnDevice)
+		try? await updating {
+			$0.header.lastUsedOnDevice = lastUsedOnDevice
+			$0.header.creatingDevice.description = deviceDescription
+		}
+	}
+}
+
+extension DeviceInfo {
+	public static func deviceDescription(
+		name: String,
+		model: String
+	) -> String {
+		"\(model) (\(name))"
 	}
 }
 
+// MARK: - ProfileStore.Error
 extension ProfileStore {
-	// Changes the currently used ProfileSnapshot, usually to one from a backup or to one just created.
-	func changeProfileSnapshot(to profileSnapshot: ProfileSnapshot) async throws {
-		var profileSnapshot = profileSnapshot
-		profileSnapshot.changeCurrentToMainnetIfNeeded()
-		try await claimProfileSnapshot(&profileSnapshot)
-		updateHeader(&profileSnapshot)
-
-		// Save the updated snapshot.
-		// Do not check the ownership since the device did claim the profile ownership.
-		try await saveProfileSnapshot(profileSnapshot, checkOwnership: false)
-		// Update to new active profile id, so it is used from now on.
-		await userDefaultsClient.setActiveProfileID(profileSnapshot.header.id)
-
-		// Update the state with the imported snapshot
-		try changeState(to: .persisted(.init(snapshot: profileSnapshot)))
-	}
-
-	/// Claim the profile by updating **lastUsedOnDevice**
-	func claimProfileSnapshot(_ snapshot: inout ProfileSnapshot) async throws {
-		snapshot.header.lastUsedOnDevice = try await Self.createDeviceInfo()
-		do {
-			try await secureStorageClient.saveDeviceIdentifierIfNeeded(snapshot.header.lastUsedOnDevice.id)
-		} catch {
-			loggerGlobal.critical("Failed to save newly generated device identifier, error: \(error)")
+	public enum Error: String, Swift.Error, Equatable {
+		case profileIDMismatch
+		case profileUsedOnAnotherDevice
+	}
+}
+
+// MARK: "Private"
+extension ProfileStore {
+	func _lens<Property>(
+		_ transform: @escaping @Sendable (Profile) -> Property?
+	) -> AnyAsyncSequence<Property> where Property: Sendable & Equatable {
+		profileSubject.compactMap(transform)
+			.share() // Multicast
+			.removeDuplicates()
+			.eraseToAnyAsyncSequence()
+	}
+
+	func _update(profile: Profile) throws {
+		try saveProfileAfterUpdateItsHeader(profile)
+	}
+}
+
+// MARK: Private
+extension ProfileStore {
+	/// Asserts identity and ownership of a profile, then updates its header, saves it and emits an update.
+	/// - Parameter updated: Profile to save (after updating its header).
+	private func saveProfileAfterUpdateItsHeader(
```

Comment:
The name of the function is not consistent with its content, the header is updated first after that the profile is saved. Could this function name be "just" "saveProfile"?

Reply:
so name and content is consistent...? FIRST update header, then profile. Name says "saveProfile AFTER Update header" right?

I think `saveProfiel` is hiding to much, more happens... `updateWallet:withProfile` ? hmm

Reply:
to be consistent with the content, shouldn't it be named then `saveProfileAfterUpdatingItsHeader`? emphasis on `Update` vs `Updating`. In current form I read it as "save profile, after that update its header"

Reply:
sure!